plugins/devtools-backend/src/api/DevToolsBackendApi.ts

```diff
@@ -0,0 +1,230 @@
+import parseArgs from 'minimist';
+import ping from 'ping';
+import { resolve as resolvePath } from 'path';
+import os from 'os';
+import fs from 'fs-extra';
+import { Lockfile } from '../util/Lockfile';
+import { memoize } from 'lodash';
+
+/** @public */
+export class DevToolsBackendApi {
+  public constructor(
+    private readonly logger: Logger,
+    private readonly config: Config,
+  ) {}
+
+  public async listExternalDependencyDetails(): Promise<ExternalDependency[]> {
+    const result: ExternalDependency[] = [];
+
+    const endpoints = this.config.getOptional<Endpoint[]>(
+      'devTools.externalDependencies.endpoints',
+    );
+    if (!endpoints) {
+      // No external dependency endpoints configured
+      return result;
+    }
+    for (const endpoint of endpoints) {
+      this.logger?.info(
+        `Checking external dependency "${endpoint.name}" at "${endpoint.target}"`,
+      );
+
+      switch (endpoint.type) {
+        case 'ping': {
+          const pingResult = await this.pingExternalDependency(endpoint);
+          result.push(pingResult);
+          break;
+        }
+        case 'fetch': {
+          const fetchResult = await this.fetchExternalDependency(endpoint);
+          result.push(fetchResult);
+          break;
+        }
+        default:
+          return result;
+      }
+    }
+
+    return result;
+  }
+
+  private async fetchExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    let status;
+    let error;
+
+    await fetch(endpoint.target)
+      .then(res => {
+        status = res.status === 200 ? 'Healthy' : 'Unhealthy';
+        this.logger.info(
+          `Fetch for ${endpoint.name} resulted in status code "${res.status}"`,
+        );
+      })
+      .catch((err: Error) => {
+        this.logger.error(`Fetch failed for ${endpoint.name} - ${err.message}`);
+        error = err.message;
+      });
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: status ?? 'Unhealthy',
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  private async pingExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    const pingResult = await ping.promise.probe(endpoint.target);
+
+    let error;
+    if (
+      pingResult.packetLoss === '100.000' ||
+      pingResult.packetLoss === 'unknown'
+    ) {
+      this.logger.error(
+        `Ping failed for ${endpoint.name} - ${pingResult.output}`,
+      );
+      error =
+        pingResult.output === ''
+          ? `${endpoint.target} - Unknown`
+          : pingResult.output;
+    }
+
+    this.logger.info(`Ping results for ${endpoint.name}: ${pingResult.output}`);
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: pingResult.alive ? 'Healthy' : 'Unhealthy',
```

Comment:
wonder if this should be a boolean, or an enum?

as it stands, you've hardcoded the string `'Unhealthy'` here and again up on line 106 (possibly others). i'd assume other places, as well. wonder if having  just a variable true/false at it's MOST simple, or if we want the flexibility, an enum which can be extended over time  (healthy, recovering, unhealthy, etc) without using strings might be valuable?

just a thought!  not a requirement :)

Reply:
Yeah, an `enum` might be nicer and I'll work on that, good suggestion

Reply:
I've changed this to use an `enum` instead 👍